src/common/test-webcertificate.c

```diff
@@ -94,13 +94,30 @@ test_locate (void)
   rmdir (workdir);
 }
 
+static void
+test_keypath (void)
+{
+  char *path;
+
+  path = cockpit_certificate_key_path ("/etc/cockpit/ws-certs.d/50-good.cert");
+  g_assert_cmpstr (path, ==, "/etc/cockpit/ws-certs.d/50-good.key");
+  g_free (path);
+  path = cockpit_certificate_key_path ("a.cert");
+  g_assert_cmpstr (path, ==, "a.key");
+  g_free (path);
+  path = cockpit_certificate_key_path ("a.crt");
+  g_assert_cmpstr (path, ==, "a.key");
+  g_free (path);
+}
```

Comment:
You could test the aborting case with `fork()` if you really wanted... Not saying you should...